polkadot/runtime/parachains/src/paras/mod.rs

```diff
@@ -1491,6 +1611,25 @@ impl<T: Config> Pallet<T> {
 		T::DbWeight::get().reads_writes(1 + pruning_tasks_done, 2 * pruning_tasks_done)
 	}
 
+	/// This function removes authorizations that have expired,
+	/// meaning their `expire_at` block is less than or equal to the current block (`now`).
+	fn prune_expired_authorizations(now: BlockNumberFor<T>) -> Weight {
+		let mut weight = T::DbWeight::get().reads(1);
+		let to_remove = AuthorizedCodeHash::<T>::iter().filter_map(|(para, (_, expire_at))| {
```

Comment:
noting that this is a good design only because we assume this map can only be populated by privileged origin, otherwise iterating the whole map to find the stale keys would not be great.